File: src/main/java/ru/mdashlw/rankedwho/player/impl/NickedPlayer.java
package ru.mdashlw.rankedwho.player.impl;

import net.minecraft.util.ChatComponentText;
import net.minecraft.util.ChatStyle;
import net.minecraft.util.EnumChatFormatting;
import net.minecraft.util.IChatComponent;
import ru.mdashlw.rankedwho.player.Player;

public class NickedPlayer implements Player {
    private String name;

    public NickedPlayer(String name) {
        this.name = name;
    }

    @Override
    public IChatComponent getMessage(int team) {
        return new ChatComponentText("Team #" + team + ": ")
                .setChatStyle(new ChatStyle()
                        .setColor(EnumChatFormatting.WHITE))
                .appendSibling(new ChatComponentText(name)
                        .setChatStyle(new ChatStyle()
                                .setColor(EnumChatFormatting.GRAY)))
                .appendSibling(new ChatComponentText(" NICKED")
                        .setChatStyle(new ChatStyle()
                                .setColor(EnumChatFormatting.RED)));
    }
}

File: src/main/kotlin/ru/mdashlw/rankedwho/player/impl/NickedPlayer.kt
package ru.mdashlw.rankedwho.player.impl

import ru.mdashlw.rankedwho.managers.NickedManager
import ru.mdashlw.rankedwho.player.Player

class NickedPlayer(override val displayname: String) : Player {
    override val rating: Int = -1
    override val kit: String = ""
    override val level: Int = 0

    override fun generateMessage(team: Int): String =
        "§fTeam #$team: §7$displayname §cNICKED"

    companion object {
        fun create(nick: String): Player {
            val name = NickedManager.get(nick)
                ?: return NickedPlayer(nick)

            return RegularPlayer.createNicked(name, nick)
        }
    }
}
